zilliqa/src/consensus.rs

```diff
@@ -306,23 +306,59 @@ impl Consensus {
             self.view = view;
             self.block_store.request_block_by_view(view)?;
         }
-        self.view = to + 1;
+
+        if to > self.view {
+            info!(
+                "*** Downloading missing blocks...view is going from {} to {}",
+                self.view,
+                to + 1
+            );
+            self.view = to + 1;
+        }
 
         Ok(())
     }
 
-    pub fn timeout(&mut self) -> Result<(PeerId, NewView)> {
+    pub fn timeout(&mut self, peers_joined: bool) -> Option<(PeerId, ExternalMessage)> {
+        if self.view == 1 {
+            if self.config.allow_single_node_network || peers_joined {
```

Comment:
IMO we should remove `peers_joined` along with the `allow_single_node_network` configuration and just always execute this block.

Reply:
righto